applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -0,0 +1,338 @@
+
+        for process in self._GetListOfProcesses():
+            process.ExecuteBeforeSolutionLoop()
+
+    def InitializeSolutionStep(self):
+
+        # Since no substepping is implemented yet, structure and fluid step/time must match
+        step_fluid = self.fluid_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        step_structure = self.structure_main_model_part.ProcessInfo[KratosMultiphysics.STEP]
+        if step_fluid != step_structure:
+            err_msg =  'Fluid step: '+ str(step_fluid) + '\n'
+            err_msg += 'Structure step: '+ str(step_structure) + '\n'
+            err_msg += 'No substepping has been implemented yet. Fluid and structure step must match.'
+            raise Exception(err_msg)
+
+        if self.is_printing_rank:
+            Kratos.Logger.PrintInfo(self._GetSimulationName(),"STEP = ", step_fluid)
+            Kratos.Logger.PrintInfo(self._GetSimulationName(),"TIME = ", self.time)
+
+        self.ApplyBoundaryConditions() #here the processes are called
+        self.ChangeMaterialProperties() #this is normally empty
+        self._GetSolver().InitializeSolutionStep()
+
+    def OutputSolutionStep(self):
+
+        has_output = self.fluid_has_output or self.structure_has_output
+        is_output_step = self.fluid_output.IsOutputStep() or self.structure_output.IsOutputStep()
+
+        if has_output and is_output_step:
+
+            for process in self._GetListOfProcesses():
+                process.ExecuteBeforeOutputStep()
+
+            if self.fluid_output.IsOutputStep():
+                self.fluid_output.PrintOutput()
+
+            if self.structure_output.IsOutputStep():
+                self.structure_output.PrintOutput()
+
+            for process in self._GetListOfProcesses():
+                process.ExecuteAfterOutputStep()
+
+        if self.save_restart:
+            fluid_restart_utility = self._GetFluidRestartUtility()
+            structure_restart_utility = self._GetStructureRestartUtility()
+            fluid_restart_utility.SaveRestart()
+            structure_restart_utility.SaveRestart()
+
+    def _CreateSolver(self):
+        import python_solvers_wrapper_fsi
+        return python_solvers_wrapper_fsi.CreateSolver(self.model, self.project_parameters)
+
+    def _GetSimulationName(self):
+        fluid_simulation_name = self.project_parameters["fluid_solver_settings"]["problem_data"]["problem_name"].GetString()
+        structure_simulation_name = self.project_parameters["structure_solver_settings"]["problem_data"]["problem_name"].GetString()
+
+        if fluid_simulation_name != structure_simulation_name:
+            err_msg =  'Different fluid and structure simulation names:\n'
+            err_msg += '\t-Fluid simulation name:' + fluid_simulation_name + '\n'
+            err_msg += '\t-Structure simulation name:' + structure_simulation_name + '\n'
+            raise Exception(err_msg)
+
+        return fluid_simulation_name
+
+    def _SetUpListOfProcesses(self):
```

Comment:
This function seems to hardcode the "old" way of defining the processes (=> without the encapsulation in "processes")
This is a bit limiting in my eyes but your call ofc